services/api.py

```diff
@@ -685,6 +687,15 @@ def get_queryset(self):
             queryset = queryset.prefetch_related('service_details__service')
 
         filters = self.request.query_params
+
+        # query parameter validation
+        f = {k: v[0].split(',') for k, v in filters.lists()}
```

Comment:
Two points.

First, we need to make sure that a comma is never part of a valid value. See SearchViewset for more info. It might be safer to only split the filters that explicitly support commas-as-separators.

Second, using the lists() method is a good idea, but it makes me realize that we should probably begin to support also  the &a=1&a=2&a=3 style for filters which support multiple values. That is, you could write something like

    {k: set([item for sublist in [x.split(',') for x in v] for item in sublist]) for k, v in d.items()}

But still the split(',') should only be done for supported fields.

This way, you could easily validate the fact that the API user has accidentally duplicated some parameter even though we don't support it (for that parameter).

Reply:
Also, you should only do the splitting once. Now it's done once for the validation and once below for the actual filter implementations.